profitbricks/src/main/java/org/jclouds/profitbricks/domain/LoadBalancer.java

```diff
@@ -0,0 +1,398 @@
+      return new AutoValue_LoadBalancer(id, name, loadBalancerAlgorithm, dataCenterId, dataCenterVersion, internetAccess, ip, lanId, state, creationTime, lastModificationTime,
+              balancedServers != null ? ImmutableList.copyOf(balancedServers) : ImmutableList.<Server>of(),
+              firewalls != null ? ImmutableList.copyOf(firewalls) : ImmutableList.<Firewall>of());
+   }
+
+   public static Builder builder() {
+      return new Builder();
+   }
+
+   public static class Builder {
+
+      public String id;
+
+      public String name;
+
+      public LoadbalancerAlgorithm loadBalancerAlgorithm;
+
+      public String dataCenterId;
+
+      public String dataCenterVersion;
+
+      public boolean internetAccess;
+
+      public String ip;
+
+      public String lanId;
+
+      public ProvisioningState state;
+
+      @Nullable
+      public Date creationTime;
+
+      public Date lastModificationTime;
+
+      public List<Server> balancedServers;
+
+      public List<Firewall> firewalls;
+
+      public Builder id(String id) {
+         this.id = id;
+         return this;
+      }
+
+      public Builder name(String name) {
+         this.name = name;
+         return this;
+      }
+
+      public Builder loadBalancerAlgorithm(LoadbalancerAlgorithm loadBalancerAlgorithm) {
+         this.loadBalancerAlgorithm = loadBalancerAlgorithm;
+         return this;
+      }
+
+      public Builder dataCenterId(String dataCenterId) {
+         this.dataCenterId = dataCenterId;
+         return this;
+      }
+
+      public Builder dataCenterVersion(String dataCenterVersion) {
+         this.dataCenterVersion = dataCenterVersion;
+         return this;
+      }
+
+      public Builder internetAccess(boolean internetAccess) {
+         this.internetAccess = internetAccess;
+         return this;
+      }
+
+      public Builder ip(String ip) {
+         this.ip = ip;
+         return this;
+      }
+
+      public Builder lanId(String lanId) {
+         this.lanId = lanId;
+         return this;
+      }
+
+      public Builder creationTime(Date creationTime) {
+         this.creationTime = creationTime;
+         return this;
+      }
+
+      public Builder state(ProvisioningState state) {
+         this.state = state;
+         return this;
+      }
+
+      public Builder lastModificationTime(Date lastModificationTime) {
+         this.lastModificationTime = lastModificationTime;
+         return this;
+      }
+
+      public Builder balancedServers(List<Server> balancedServers) {
+         this.balancedServers = balancedServers;
+         return this;
+      }
+
+      public Builder firewalls(List<Firewall> firewalls) {
+         this.firewalls = firewalls;
+         return this;
+      }
+
+      public LoadBalancer build() {
+         return LoadBalancer.create(id, name, loadBalancerAlgorithm, dataCenterId, dataCenterVersion, internetAccess, ip, lanId, state, creationTime, lastModificationTime, balancedServers, firewalls);
```

Comment:
Could you add validation for the IP here? See this [from nic](https://github.com/devcsrj/jclouds-labs/blob/loadbalancer-api/profitbricks/src/main/java/org/jclouds/profitbricks/domain/Nic.java#L191) for reference